operators/endpointmetrics/controllers/lease/lease_controller.go

```diff
@@ -0,0 +1,167 @@
+package lease
```

Comment:
why do we need this?
The lease is handled by pkg/util/lease.go now. If want to fix the hub kubeconfig roated problem, should we fix it in the lease.go?

Reply:
`lease.go` was using a utility function `open-cluster-management.io/addon-framework/pkg/lease.NewLeaseUpdater`  earlier, and this did not allow for refreshing hub kubeconfig. In the newly refactored code, lease.go is copied into the project, modified to take a path to hub kubeconfig so it can be refreshed when hub kubeconfig becomes invalid. Do you see a simpler way to address this?

Reply:
no, just want to confirm this is the best option recommended by foundation squad